test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/EnumTypeModelBinderTest.cs

```diff
@@ -0,0 +1,285 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Binders;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding
+{
+    public class EnumTypeModelBinderTest
+    {
+        [Theory]
+        [InlineData(typeof(IntEnum?))]
+        [InlineData(typeof(FlagsEnum?))]
+        public async Task BindModel_SetsModel_ForNullableEnumTypes(Type modelType)
+        {
+            // Arrange
+            var bindingContext = GetBindingContext(modelType);
+            bindingContext.ValueProvider = new SimpleValueProvider
+            {
+                { "theModelName", "" }
+            };
+
+            var binder = new EnumTypeModelBinder(new MvcOptions());
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(bindingContext.Result.IsModelSet);
+            Assert.Null(bindingContext.Result.Model);
+        }
+
+        [Theory]
+        [InlineData(typeof(IntEnum))]
+        [InlineData(typeof(FlagsEnum))]
+        public async Task BindModel_AddsErrorToModelState_ForEmptyValue(Type modelType)
+        {
+            // Arrange
+            var message = "The value '' is invalid.";
+            var bindingContext = GetBindingContext(modelType);
+            bindingContext.ValueProvider = new SimpleValueProvider
+            {
+                { "theModelName", "" }
+            };
+
+            var binder = new EnumTypeModelBinder(new MvcOptions());
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.False(bindingContext.Result.IsModelSet);
+            Assert.Null(bindingContext.Result.Model);
+            Assert.False(bindingContext.ModelState.IsValid);
+            var error = Assert.Single(bindingContext.ModelState["theModelName"].Errors);
+            Assert.Equal(message, error.ErrorMessage);
+        }
+
+        [Fact]
+        public async Task BindModel_BindsEnumModels_IfArrayElementIsStringKey()
+        {
+            // Arrange
+            var bindingContext = GetBindingContext(typeof(IntEnum));
+            bindingContext.ValueProvider = new SimpleValueProvider
+            {
+                { "theModelName", new object[] { "Value1" } }
+            };
+
+            var binder = new EnumTypeModelBinder(new MvcOptions());
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.True(bindingContext.Result.IsModelSet);
+            var boundModel = Assert.IsType<IntEnum>(bindingContext.Result.Model);
+            Assert.Equal(IntEnum.Value1, boundModel);
+        }
+
+        [Fact]
+        public async Task BindModel_BindsEnumModels_IfArrayElementIsStringValue()
```

Comment:
Doesn't this do the same thing as `BindModel_BindsIntEnumModels` when `allowBindingUndefinedValueToEnumType` has the default value? Suggest removing this test if so.